docs/source/conf-schemas.rst

```diff
@@ -4,188 +4,387 @@ Schemas
-      "eventType": "string",
-      "responseElements": "string",
-      "awsRegion": "string",
-      "eventName": "string",
-      "userIdentity": "string",
-      "eventSource": "string",
-      "requestID": "string",
-      "apiVersion": "string",
-      "userAgent": "string",
-      "sourceIPAddress": "string",
-      "recipientAccountId": "string"
-    },
-    "hints" : {
-      "records": "Records[*]"               # JSONPath to records
+They represent the structure of a given log in the form of key/value pairs.
+
+Each key in a schema corresponds to the name of a field referenced by rules.  Its value is data type the field is cast to.
+
+Ordering is strict.
+
+Example log schema::
+
+  {
+    "example_log_name": {
+      "parser": "parser-name-goes-here",
+      "schema": {
+        "field_1": "string",
+        "field_2": "integer",
+        "field_3": "boolean",
+        "field_4": "float",
+        "field_5": [],
+        "field_6": {}
+      }
     }
-  },
-  ....
+  }
 
+Example rule:
 
-Here are the basics:
+.. code-block:: python
 
-* Keys defined in the ``schema`` must exist
-* Values are weakly/loosely typed and enforced
-* Supported types: ``string``, ``integer``, ``boolean``, ``float``, ``{}``, and ``[]``
-* An empty array (``[]``) imply zero or more elements of any type
-* An empty map (``{}``) implies zero or more key/value pairs
-* Schemas can be as tight or as loose as you want (see Example: osquery)
-* If you have nested records, you can specify a JSONPath to your ``records`` with ``hints``
+  @rule(logs=['example_log_name'],              # the log_name as defined above
+        matchers=[],
+        outputs=['slack'])
+  def example_rule(rec):
+    """Description of the rule"""
+    return (
+      rec['field_1'] == 'string-value' and      # fields as defined in the schema above
+      rec['field_2'] < 5 and
+      'random-key-name' in rec['field_6']
+    )
 
-JSON Example: Inspec
---------------------
-Schema::
 
-  "inspec": {
-    "schema": {
-      "title": "string",
-      "desc": "string",
-      "impact": "string",
-      "refs": "string",
-      "tags": "string",
-      "code": "string",
-      "id": "string",
-      "source_location": "string",
-      "results": "string"
-    },
-    "parser": "json",
-    "hints": {
-      "records": "profiles[*].controls[*]"
+Options
+-------
+
+=============     ========     ======================
+Key               Required     Description
+-------------     ---------    ----------
+``parser``        ``true``     The name of the parser to use for a given log's data-type.   Options include ``json, json-gzip, csv, kv, or syslog``
+``schema``        ``true``     A map of key/value pairs of the name of each field with its type
+``hints``         ``false``    A map of key/value pairs to assist in classification
+``configuration`` ``false``    Options for nested types, delimiters/separators, and more
+=============     =========    ======================
```

Comment:
This table is not rendering in GitHub. Just wanted to double-check that it works in the final view (whatever that is)